unittest/gunit/keyring_vault/vault_io-t.cc

```diff
@@ -0,0 +1,641 @@
+#include <my_global.h>
+#include <gtest/gtest.h>
+#include <fstream>
+#include "mock_logger.h"
+#include "vault_io.h"
+#include "incorrect_vault_key.h"
+#include <string.h>
+#include <curl/curl.h>
+#include "generate_credential_file.h"
+#include "uuid.h"
+#include "vault_mount.h"
+
+#if defined(HAVE_PSI_INTERFACE)
+namespace keyring
+{
+  PSI_memory_key key_memory_KEYRING = PSI_NOT_INSTRUMENTED;
+}
+#endif
+
+namespace keyring__vault_io_unittest
+{
+  using namespace keyring;
+
+  using ::testing::Return;
+  using ::testing::StrEq;
+  using ::testing::_;
+  using ::testing::SetArgPointee;
+
+  static std::string uuid = generate_uuid();
+  CURL *curl = NULL;
+  std::string credential_file_url = "./keyring_vault.conf";
+  ILogger *logger;
+
+  class Vault_io_test : public ::testing::Test
+  {
+  protected:
+    virtual void SetUp()
+    {
+      ASSERT_TRUE(curl != NULL);
+      vault_curl = new Vault_curl(logger, curl);
+      vault_parser = new Vault_parser(logger);
+    }
+
+  protected:
+    IVault_curl *vault_curl;
+    IVault_parser *vault_parser;
+  };
+
+  TEST_F(Vault_io_test, InitWithNotExisitingCredentialFile)
+  {
+    std::string credential_file_name("./some_funny_name");
+    Vault_io vault_io(logger, vault_curl, vault_parser);
+    remove(credential_file_name.c_str());
+    EXPECT_CALL(*(reinterpret_cast<Mock_logger*>(logger)),
+      log(MY_ERROR_LEVEL, StrEq("Could not open file with credentials.")));
+    EXPECT_TRUE(vault_io.init(&credential_file_name));
+
+    remove(credential_file_name.c_str());
+  }
+
+  TEST_F(Vault_io_test, InitWithInvalidToken)
+  {
+    std::string conf_with_invalid_token("./invalid_token.conf");
+    ASSERT_FALSE(generate_credential_file(conf_with_invalid_token, WITH_INVALID_TOKEN));
+
+    Vault_io vault_io(logger, vault_curl, vault_parser);
+    EXPECT_FALSE(vault_io.init(&conf_with_invalid_token));
+
+    EXPECT_CALL(*(reinterpret_cast<Mock_logger*>(logger)),
+      log(MY_ERROR_LEVEL, StrEq("Could not retrieve list of keys from Vault. "
+          "Vault has returned the following error(s): [\"permission denied\"]")));
+    ISerialized_object *serialized_keys= NULL;
+    EXPECT_TRUE(vault_io.get_serialized_object(&serialized_keys));
+  }
+
+  TEST_F(Vault_io_test, GetSerializedObjectWithTwoKeys)
+  {
+    Vault_io vault_io(logger, vault_curl, vault_parser);
+
+    EXPECT_FALSE(vault_io.init(&credential_file_url));
+
+    // First add two keys into Vault
+    Vault_key key1((uuid+"key1").c_str(), "AES", "Arczi", "Artur", 5);
```

Comment:
Hav you considered extracting "(uuid+"key1").c_str()" in all its forms to variables? Spaces around +